cmd/boots/main.go

```diff
@@ -97,13 +140,162 @@ func main() {
 		}
 	}()
 
-	mainlog.Info("serving tftp")
-	go ServeTFTP()
-	mainlog.Info("serving dhcp")
-	go ServeDHCP()
+	var nextServer net.IP
+	var httpServerFQDN string
+	g, ctx := errgroup.WithContext(ctx)
+	lg := defaultLogger(cfg.logLevel)
+	lg = lg.WithValues("service", "github.com/tinkerbell/boots")
+	lg = lg.WithName("github.com/tinkerbell/ipxedust")
+	ipxe := &ipxedust.Server{
+		Log:                  lg,
+		EnableTFTPSinglePort: true,
+		TFTP:                 ipxedust.ServerSpec{Disabled: true},
+		HTTP:                 ipxedust.ServerSpec{Disabled: true},
+	}
+
+	if cfg.remoteTFTPAddr == "" { // use local iPXE binary service for TFTP
+		if !cfg.iTFTPDisabled {
```

Comment:
imo this would also be easier to read as:
```go
if cfg.iPXETFTPEnabled {
...
```

Reply:
updated